src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -87,4 +89,16 @@ public int getTotalMessageCount(){
     PreparedQuery results = datastore.prepare(query);
     return results.countEntities(FetchOptions.Builder.withLimit(1000));
   }
+
+  /** Returns the total number of users with messages. */
+  public int getTotalUserCount(){
+    Query query = new Query("Message");
+    PreparedQuery results = datastore.prepare(query);
+    Set users = new HashSet();
+    for (Entity message : results.asList(FetchOptions.Builder.withLimit(1000))){
```

Comment:
(i know this is part of the sample code, just wanted to get your thoughts)

what happens when there are more than 1000 messages? 
what happens if we do not specify this limit?

Reply:
That's actually something I was reading up a bit on because I found the FetchOptions thing very cumbersome. I'm pretty sure if there are more messages than the limit, only the first 1000 would be fetched making the stats inaccurate. I have also tried using `.withDefaults()` which according to documentation sets all values to `null` and it seems to work as well. I can't find the documentation for the behavior when `limit` is `null`, but if I were to guess I would guess that it fetches everything with no limit.